socorro/scripts/fetch_crash_data.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import argparse
+import json
+import os
+import os.path
+
+import requests
+
+from socorro.lib.datetimeutil import JsonDTEncoder
+from socorro.scripts import WrappedTextHelpFormatter
+
+
+EPILOG = """
+Given a crash id, fetches crash data and puts it in specified directory
+
+This requires an auth-token to be in the environment::
+
+    SOCORRO_API_TOKEN=xyz
+
+To create an API token for Socorro in -prod, visit:
+
+    https://crash-stats.mozilla.com/api/tokens/
+
+"""
+
+
+class CrashDoesNotExist(Exception):
+    pass
+
+
+def create_dir_if_needed(d):
+    if not os.path.exists(d):
+        os.makedirs(d)
+
+
+def fetch_crash(outputdir, api_token, crash_id):
+    headers = {
+        'Auth-Token': api_token
+    }
+
+    # Fetch raw crash metadata
+    print('Fetching %s' % crash_id)
+    resp = requests.get(
+        'https://crash-stats.mozilla.com/api/RawCrash/',
+        params={
+            'crash_id': crash_id,
+            'format': 'meta',
+        },
+        headers=headers,
+    )
+    if resp.status_code == 404:
+        raise CrashDoesNotExist(crash_id)
+
+    raw_crash = resp.json()
+
+    # Fetch dumps from -prod
+    dumps = {}
+    dump_names = raw_crash.get('dump_checksums', {}).keys()
+    for dump_name in dump_names:
+        print('Fetching %s -> %s' % (crash_id, dump_name))
+        if dump_name == 'upload_file_minidump':
+            dump_name = 'dump'
+
+        resp = requests.get(
+            'https://crash-stats.mozilla.com/api/RawCrash/',
+            params={
+                'crash_id': crash_id,
+                'format': 'raw',
+                'name': dump_name
+            },
+            headers=headers,
+        )
+
+        if resp.status_code != 200:
+            raise Exception('Something unexpected happened. status_code %s, content %s' % (
+                resp.status_code, resp.content)
+            )
```

Comment:
I want the content, too. In the case of errors, we might get an error message that clarifies the issue.